src/session.ts

```diff
@@ -599,6 +602,32 @@ export class SessionManager implements Middleware {
                                         ? `${this.versionDetails.displayVersion} (${this.versionDetails.architecture})`
                                         : this.versionDetails.displayVersion,
                                     SessionStatus.Running);
+
+                                // If the user opted to not check for updates, then don't.
+                                if (!this.sessionSettings.promptToUpdatePowerShell) { return; }
+
+                                try {
+                                    const localVersion = semver.parse(this.versionDetails.version);
+                                    if (semver.lt(localVersion, "6.0.0")) {
+                                        // Skip prompting when using Windows PowerShell for now.
+                                        return;
+                                    }
+
+                                    // Fetch the latest PowerShell releases from GitHub.
+                                    let release: GitHubReleaseInformation;
+                                    if (semver.prerelease(localVersion)) {
+                                        // This gets all releases and the first one is the latest prerelease if
+                                        // there is a prerelease version.
+                                        release = await GitHubReleaseInformation.FetchLatestRelease(true);
+                                    } else {
+                                        release = await GitHubReleaseInformation.FetchLatestRelease(false);
+                                    }
+
+                                    await InvokePowerShellUpdateCheck(
+                                        localVersion, this.versionDetails.architecture, release);
+                                } catch (e) {
```

Comment:
Variable not needed any more since TS 2.5 apparently: https://mariusschulz.com/blog/optional-catch-binding-in-typescript
```suggestion
                                } catch {
```